ipamutils/utils.go

```diff
@@ -115,7 +115,7 @@ func splitNetwork(size int, base *net.IPNet) []*net.IPNet {
 
 	for i := 0; i < n; i++ {
 		ip := copyIP(base.IP)
-		addIntToIP(ip, uint(i<<s))
+		addIntToIP(ip, uint(i), s)
```

Comment:
Some observation:
function splitNetwork: Given a network address range "base", and a desired subnet lengh "size", return all disjointed subnet with length "size" within network address range "base".
For instance, if base=10.10.0.0/16, and size=24, returned subnets 10.10.1.0/24, 10.10.2.0/24 ... 10.10.255.0/24

The function works fine for ipv4(int32). For ipv6 (8 bytes) two things should be watched for
1. numbers of computed subnets may exceeds int32/64,(see line 112)  for instance, base=fc00::/8 , size 120. there are 1<<(120-8) number of subnets. We perhaps should reject this kind of request.
2. the host portion of subnet is bigger than int32/64; for instance with base=fdc00::/8, size 12,  line 118 becomes s=128-12=116, and (1<<116) > any integer storage. This is legit bug that perhaps can be addressed with a fix.